### PR TITLE
fix(cli): don't require `--id` when enabling a deploy key

### PR DESCRIPTION
**Pull request by JohnVillalovos**
No longer require `--id` when doing:
  gitlab project-key enable

Now only the --project-id and --key-id are required.

### CODE REVIEW

**Review by nejch — COMMENTED** [review]
> Hi @nejch
> 
> Maybe there is a simpler way to do this.
> 
> As an FYI. I have the idea, as a follow-up, to add a `help` argument to the `CustomAction` and `register_custom_action` to enable adding help text to commands if desired

I took a look and looks like the best way to do it. I can't think of another way since custom actions can be really different sometimes :bow: Just 2 nits from my side as it's a big PR but good to go otherwise!

File: gitlab/cli.py [hunk 1]
```diff
@@ -29,12 +30,21 @@
 camel_upperlower_regex = re.compile(r"([A-Z]+)([A-Z][a-z])")
 camel_lowerupper_regex = re.compile(r"([a-z\d])([A-Z])")
 
+
+@dataclasses.dataclass
+class CustomAction:
+    required: Tuple[str, ...]
+    optional: Tuple[str, ...]
+    in_object: bool
+    id_argument: bool  # if the `_id_attr` value should be a required argument
```
[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply gitlab/cli.py]: Good idea. That's a better name. Done.

File: gitlab/cli.py [hunk 2]
```diff
@@ -72,10 +82,12 @@ def format_help(self) -> str:
 
 
 def register_custom_action(
+    *,
     cls_names: Union[str, Tuple[str, ...]],
-    mandatory: Tuple[str, ...] = (),
+    required: Tuple[str, ...] = (),
     optional: Tuple[str, ...] = (),
     custom_action: Optional[str] = None,
+    id_argument: bool = True,  # if the `_id_attr` value should be a required argument
```
[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply gitlab/cli.py]: Done, but in this PR.

I created the `.git-blame-ignore-revs` in a commit.

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Require keyword arguments for register_custom_action
+19251506f734117a564c59e157221f37e5cf97ef
```
Comment (nejch) [inline .git-blame-ignore-revs:2]: Moved to d74545a309ed02fdc8d32157f8ccb9f7559cd185 after rebase, this should hopefully stay after merge:

```suggestion
d74545a309ed02fdc8d32157f8ccb9f7559cd185
```

**Review by nejch — APPROVED** [review]

**Pull request merged**